### ISSUE TITLE
The code only works on num_queries=1

### ISSUE THREAD

**Issue by IRNLPCoder** [megvii-research/DPGN#5]
https://github.com/megvii-research/DPGN/blob/b9401115b43319544c07718890e115e7f4fc2a1d/main.py#L316

`self.pred_loss(query_node_pred_generation, query_label.long()).mean()`

For example, 5 way 5 shot, num_queries=1, 
`query_node_pred_generation` has shape of [batch_size, 5, 5]
`query_label` has shape of [batch_size, 5]

5 way 5 shot, num_queries=2, 
`query_node_pred_generation` have shape of [batch_size, 10, 5]
`query_label` has shape of [batch_size, 10]

In `query_node_pred_generation`, which dimension is the class (i.e., N ways)?

**Comment by zilunzhang** [comment]
> https://github.com/megvii-research/DPGN/blob/b9401115b43319544c07718890e115e7f4fc2a1d/main.py#L316
> 
> `self.pred_loss(query_node_pred_generation, query_label.long()).mean()`
> 
> For example, 5 way 1 shot, num_queries=1,
> `query_node_pred_generation` has shape of [batch_size, 5, 5]
> `query_label` has shape of [batch_size, 5]
> 
> 5 way 1 shot, num_queries=2,
> `query_node_pred_generation` have shape of [batch_size, 10, 5]
> `query_label` has shape of [batch_size, 10]
> 
> In `query_node_pred_generation`, which dimension is the class (i.e., N ways)?

Dear Jing Li,

It is very unfortunate that the codebase is designed only for the setting of ``num_queries = 1`` in the beginning. 

We are sorry that the implementation of auxiliary loss (``query_node_pred_loss``) confuses many readers. Therefore, we decided to propose a modified version that follows most people's intuition. The accuracy for the result of 5-way 1-shot, resnet12, mini-imagenet is still around 67.70±0.52 (almost the same as reported accuracy on the paper, 67.77).

 Please replace the content at line 316-319 of main.py by the code below:

```bash
        query_node_pred_loss = []
        for query_node_pred_generation in query_node_pred_generations_:
            temp_query_node_pred_generation = query_node_pred_generation.contiguous().view(-1, query_node_pred_generation.shape[-1])
            temp_query_label = query_label.long().contiguous().view(-1, )
            temp_loss = self.pred_loss(temp_query_node_pred_generation, temp_query_label).mean()
            query_node_pred_loss.append(temp_loss)
```
Also, some configs need to be revised:

``config['num_loss_generation'] = 7``
``train_opt['dec_lr'] = 18000``

Let us know if you have any other confusion when using the DPGN.

Yours,
DPGN Team

**Comment by shi1997Yee** [comment]
Thanks for your work, in your experiments, 'num_queries' is set to 1  in stead of 15. Does it meet the standard few-shot learning setting ?

**Comment by lgx12345678** [comment]
> [DPGN/main.py](https://github.com/megvii-research/DPGN/blob/b9401115b43319544c07718890e115e7f4fc2a1d/main.py#L316)
> 
> Line 316 in [b940111](/megvii-research/DPGN/commit/b9401115b43319544c07718890e115e7f4fc2a1d)
> 
>  query_node_pred_loss = [ 
> `self.pred_loss(query_node_pred_generation, query_label.long()).mean()`
> 
> For example, 5 way 1 shot, num_queries=1, `query_node_pred_generation` has shape of [batch_size, 5, 5] `query_label` has shape of [batch_size, 5]
> 
> 5 way 1 shot, num_queries=2, `query_node_pred_generation` have shape of [batch_size, 10, 5] `query_label` has shape of [batch_size, 10]
> 
> In `query_node_pred_generation`, which dimension is the class (i.e., N ways)?

Hello, I can't unable to download the 'tiered-imagenet' dataset due to permission issues. Could you please share this dataset with me?
